gradle-plugin/build.gradle

```diff
@@ -3,30 +3,43 @@ plugins {
   alias libs.plugins.gradlePublish
   alias libs.plugins.buildconfig
   id 'java-gradle-plugin'
-  id 'com.squareup.anvil.publish'
+}
+
+def pomName = 'Anvil Gradle Plugin'
+def pomDescription = 'A Kotlin compiler plugin to make dependency injection with Dagger 2 ' +
+    'easier by automatically merging Dagger modules and component interfaces.'
+
+// This module is part of two builds:
+// 1. The main build, which publishes the plugin to the Gradle plugin portal.
+// 2. The included build, which is used by the main build to test the plugin.
+// Only the main build should publish the plugin.
+if (rootProject.name == 'anvil') {
+  apply plugin: 'com.squareup.anvil.publish'
+
+  publish {
+    configurePom(
+        artifactId: 'gradle-plugin',
+        pomName: pomName,
+        pomDescription: pomDescription
+    )
+  }
+  layout.buildDirectory.set(file('build-anvil'))
+} else {
+  layout.buildDirectory.set(file('build-composite-wrapper'))
 }
 
 buildConfig {
   className('BuildProperties')
   packageName('com.squareup.anvil.plugin')
-  useKotlinOutput { topLevelConstants = true }
+  useKotlinOutput {
+    internalVisibility = true
```

Comment:
There's lots of ambiguous terms here. 🤣

This makes it so that properties in the generated `BuildProperties.kt` file definitely have internal visibility, like:

```kotlin
...
internal const val GROUP: String = "com.squareup.anvil"
...
```

This plugin generates internal properties by default, so I'm adding it here just to be explicit.

Reply:
Perhaps this also causes the error I'm seeing trying to sync the project:
`A problem occurred evaluating project ':gradle-plugin-build-logic:gradle-plugin'.
> Could not get unknown property 'GROUP' for extension 'buildConfig' of type com.github.gmazzo.gradle.plugins.internal.DefaultBuildConfigExtension.`